src/raft.c

```diff
@@ -1384,7 +1519,11 @@ static void handleReadOnlyCommand(void *arg, int can_read)
         goto exit;
     }
 
-    executeRaftRedisCommandArray(&req->r.redis.cmds, req->ctx, req->ctx);
+    RaftRedisCommandArray * cmds = &req->r.redis.cmds;
+
+    if (!redis_raft.config->sharding || validateRaftRedisCommandArray(&redis_raft, cmds, req->ctx) == RR_OK) {
```

Comment:
`redis_raft.config->sharding` is this check for shortcut circuit, like for performance? Is there way to avoid this? Because state machine becomes dependent on config, we take logs and rdb, restart in another machien without `sharding` config. We'll have hard time to understand what is going on :)

Reply:
yea, but if sharding is disabled, we might have other issues too (i.e. allow multi slot operations through)

Reply:
That applies to new operations. I'm worried about current data set. I think log replay should not rely on things that are not part of the state machine. Just easy to make mistakes.  It should be deterministic, replay should always give the same result. Just consider if we really need this check.

Reply:
ok, that makes sense.  I could do the same optimization as the slot is stored, so we can default the slot to -1 for non sharding cases and use that to short circuit.

Reply:
ok, made the change I mentioned, no longer based on configuration, based on slot == -1 which it will be by default unless handleSharding() changes it.